docs/content/testing/testing-guidelines.md

```diff
@@ -0,0 +1,252 @@
+**Warning:** The [pre-alpha](https://docs.improbable.io/reference/latest/shared/release-policy#maturity-stages) release is for evaluation purposes only, with limited documentation - see the guidance on [Recommended use](../../../README.md#recommended-use).
+
+-----
+
+# Testing guidelines
+
+The tests should:
+
+- Set up some initial state
+- Execute a method that acts upon this state and results in a new state
+- Make assertions in the new state against the expected changes
+- Clean up
+
+NUnit allows this to be done through the use of fixtures and special method annotations.
+
+## Module directory structure
+
+The Unity GDK project's assets are divided into modules, such as `Core`, `Physics`, `PlayerLifecycle`.
+
+Each module is in its own directory within `workers\unity\Assets\Gdk`.
+
+Each module has its own tests that cover the functionality of that module.
+
+Unity Test Runner also allows separate Playmode and Editmode tests. For more information on this division, please refer to the [Unity Test Runner manual page](https://docs.unity3d.com/Manual/testing-editortestsrunner.html).
+
+Each module folder will contain the tests about that module, and within separate subdirectories if a module needs both Editmode and Playmode testing - this is a requirement from Unity:
+
+> The Editmode and Playmode folders need to be separate as the tests need to be in different assemblies, and currently Unity `.asmdef` files affect all files within the same directory.
+
+Within the root of the module, the `Tests` directory should contain `Editmode`
+ and `Playmode` subdirectories, with their own assembly definition files.
+
+For example, for the `Core` module, the module files are within
+ `workers\unity\Assets\Gdk\Core`. The Editmode tests should be in
+ `workers\unity\Assets\Gdk\Core\Tests\Editmode`.
+
+Test assembly definition files must match this pattern:
+> `Improbable.Gdk.<ModuleName>.<TestMode>Tests.asmdef`
+
+If a class to be tested will have both Editmode and Playmode tests, you can name the test filenames as `<ClassName>EditmodeTests.cs` and `<ClassName>PlaymodeTests.cs` within their respective assemblies.
+
+Within each test folder, the folder structure should match the non-test code structure.
+
+If any file path can be broken down into the pattern `<module path>/<path within module>.cs`, the Editmode tests for this class should be in `<module path>/Tests/Editmode/<path within module>Tests.cs`.
```

Comment:
it feels like we repeat line 40 here. Maybe move it all together in one paragraph and make it an example of that

Reply:
I'll reorg these parts a bit